types/object-merge/index.d.ts

```diff
@@ -0,0 +1,164 @@
+     */
+    (opts: ObjectMergeOptions, ...shadows: object[]): object;
+    /**
+     * Merges JavaScript objects recursively without altering the objects merged.
+     * @author <a href="mailto:matthewkastor@gmail.com">Matthew Kastor</a>
+     * @param shadows [[shadows]...] One or more objects to merge. Each
+     *  argument given will be treated as an object to merge. Each object
+     *  overwrites the previous objects descendant properties if the property name
+     *  matches. If objects properties are objects they will be merged recursively
+     *  as well.
+     * @returns Returns a single merged object composed from clones of the
+     *  input objects.
+     * @example
+     *  var objectMerge = require('object-merge');
+     *  var x = {
+     *      a : 'a',
+     *      b : 'b',
+     *      c : {
+     *          d : 'd',
+     *          e : 'e',
+     *          f : {
+     *              g : 'g'
+     *          }
+     *      }
+     *  };
+     *  var y = {
+     *      a : '`a',
+     *      b : '`b',
+     *      c : {
+     *          d : '`d'
+     *      }
+     *  };
+     *  var z = {
+     *      a : {
+     *          b : '``b'
+     *      },
+     *      fun : function foo () {
+     *          return 'foo';
+     *      },
+     *      aps : Array.prototype.slice
+     *  };
+     *  var out = objectMerge(x, y, z);
+     *  // out.a will be {
+     *  //         b : '``b'
+     *  //     }
+     *  // out.b will be '`b'
+     *  // out.c will be {
+     *  //         d : '`d',
+     *  //         e : 'e',
+     *  //         f : {
+     *  //             g : 'g'
+     *  //         }
+     *  //     }
+     *  // out.fun will be a clone of z.fun
+     *  // out.aps will be equal to z.aps
+     */
+    (...shadows: object[]): object;
+    /**
+     * Creates a new options object suitable for use with objectMerge.
+     * @param [opts] An object specifying the options.
+     * @param [opts.depth = false] Specifies the depth to traverse objects
+     *  during merging. If this is set to false then there will be no depth limit.
+     * @param [opts.throwOnCircularRef = true] Set to false to suppress
+     *  errors on circular references.
+     * @returns Returns an instance of ObjectMergeOptions
+     *  to be used with objectMerge.
+     * @example
+     *  var opts = objectMerge.createOptions({
+     *      depth : 2,
+     *      throwOnCircularRef : false
+     *  });
+     *  var obj1 = {
+     *      a1 : {
+     *          a2 : {
+     *              a3 : {}
+     *          }
+     *      }
+     *  };
+     *  var obj2 = {
+     *      a1 : {
+     *          a2 : {
+     *              a3 : 'will not be in output'
+     *          },
+     *          a22 : {}
+     *      }
+     *  };
+     *  objectMerge(opts, obj1, obj2);
+     */
+    createOptions: (options: object) => ObjectMergeOptions;
+}
+
+declare const objectMerge: ObjectMerge;
+
+export default objectMerge;
```

Comment:
This should be [declared with `export = objectMerge` given how it's actually defined.](https://github.com/matthewkastor/object-merge/blob/48c060be94fbd91ec53a85cf0b9a56ba1a7d2ed0/src/object-merge.js#L220) It should then be imported with `import objectMerge = require('object-merge')` in this repo.

See [here](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/b1fdb2983c3f6ca0a21d60f503da30968ee2030a#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default) for some rationale on why you need to import that way.